ansible/roles/ocp4-workload-debezium-demo/tasks/provision_e2e_demo.yaml

```diff
@@ -32,35 +32,36 @@
     namespace: '{{ e2e_namespace }}'
     resource_definition: "{{ lookup('template', 'configmap-data-sql.yaml.j2') }}"
 
-- name: Deploy MS SQL Server Database 
+- name: Deploy MS SQL Server Database
   k8s:
     state: present
     namespace: '{{ e2e_namespace }}'
     resource_definition: "{{ lookup('template', 'mssql-server-linux.yaml.j2') }}"
 
-- name: Wait for Database to Start 
-  k8s_facts:
+- name: Wait for Database to Start
+  k8s_info:
     kind: Deployment
     name: mssql-server-linux
     namespace: '{{ e2e_namespace }}'
     api_version: apps/v1
-  register: mssql_resource
-  until: >-
-    (mssql_resource.resources is defined)
-    and (mssql_resource.resources|length>0)
-    and (mssql_resource.resources[0].status is defined)
-    and (mssql_resource.resources[0].status.readyReplicas is defined)
-    and (mssql_resource.resources[0].status.readyReplicas > 0)
+  register: r_k8s_info
+  until: r_k8s_info.resources[0].status.readyReplicas | default(0) > 0
```

Comment:
The main concern I have with shrinking this so much is when Ansible checks on the object and one of them doesn't exist yet. So, if it checks for the existence of the `Deployment` and it doesn't exist yet, it will fail. If the Deployment exists, but the `readyReplicas` field doesn't, it will fail. This is not to be confused with the `readyReplicas` being equal to zero. It is why we defined multiple levels to check as the objects are rolling out.

Probably less of a problem with Deployments vs something like a CSV...but this can still cause problems if the cluster is a bit slow.

Reply:
All this *and* stuff isn't even necessary. Just make it a list of *when* conditions in the order to be checked.

Reply:
starting at ansible 2.8  that shortcut works.

```yaml
- hosts: localhost
  gather_facts: false
  vars:
    a:
      b:
        c:
          d: ok
  tasks:
    - debug:
        msg: "ok"
      when: a.c.e.f.g|default(0) > 0
```

with 2.7.18, error

```TASK [debug] **************************************************************************************************************************************
Tuesday 24 November 2020  15:26:12 +0100 (0:00:00.088)       0:00:00.088 ****** 
fatal: [localhost]: FAILED! => 
  msg: |-
    The conditional check 'a.c.e.f.g|default(0) > 0' failed. The error was: error while evaluating conditional (a.c.e.f.g|default(0) > 0): 'dict object' has no attribute 'c'
  
    The error appears to have been in '/tmp/test.yaml': line 9, column 7, but may
    be elsewhere in the file depending on the exact syntax problem.
  
    The offending line appears to be:
  
      tasks:
        - debug:
          ^ here
```

with 2.8.0, it works.

I think we use at least ansible 2.9 for all our deployments, right ?

Reply:
Right. We are targeting ansible 2.9, so we can just shortcut past all these checks. It was only older versions where `default()` wouldn't catch the error on undefined.